models/target.go

```diff
@@ -0,0 +1,116 @@
+package models
+
+import (
+	"fmt"
+	"net"
+	"strings"
+
+	"github.com/miekg/dns"
+	"github.com/pkg/errors"
+)
+
+/* .Target is kind of a mess.
+For simple rtypes it is the record's value. (i.e. for an A record
+	it is the IP address).
+For complex rtypes (like an MX record has a preference and a value)
+	it might be a space-delimited string with all the parameters, or it
+	might just be the hostname.
+
+This was a bad design decision that I regret. Eventually we will eliminate this
+field and replace it with setters/getters.  The setters/getters are below
+so that it is easy to do things the right way in preparation.
+*/
+
+// GetTargetField returns the target. There may be other fields (for example
+// an MX record also has a .MxPreference field.
+func (rc *RecordConfig) GetTargetField() string {
+	return rc.Target
+}
+
+// GetTargetSingle returns the target for types that have a single value target
+// and panics for all others.
+func (rc *RecordConfig) GetTargetSingle() string {
```

Comment:
This confuses me, and appears to be unused. Whitelisting might be a better approach than blacklisting if we need to keep it.

Reply:
I've commented out a lot of these functions and only brought back the ones that are in use.